ec2/cron_script.py

```diff
@@ -15,20 +17,46 @@
 TRIGGER_FUNCTION_NAME = 'cowin-notification-service-dev-trigger_district_updates'
 UPDATE_FUNCTION_NAME = 'cowin-notification-service-dev-update_district_slots'
 
+TIME_SLEEP_FROM = time(00,00)
+TIME_TILL_ASLEEP = time(5,30)
+
 def random_str(length):
     res = ''.join(random.choices(string.ascii_uppercase + string.digits, k=length))
     return str(res)
 
+# Checking if time is between a specific range
+# https://stackoverflow.com/questions/10048249/how-do-i-determine-if-current-time-is-within-a-specified-range-using-pythons-da
+# https://www.geeksforgeeks.org/get-current-time-in-different-timezone-using-python/
+def is_time_between(begin_time, end_time, check_time=None):
+    IST = pytz.timezone('Asia/Kolkata')
+    # If check time is not given, default to current IST time
+    check_time = check_time or datetime.now(IST).time()
+    if begin_time < end_time:
+        return check_time >= begin_time and check_time <= end_time
+    else: # crosses midnight
+        return check_time >= begin_time or check_time <= end_time
+
 if __name__ == "__main__":
     while True:
         try:
-            response = client.invoke(FunctionName=TRIGGER_FUNCTION_NAME,
-                          InvocationType='Event',Payload='')
-            var = client.get_function_configuration(FunctionName=UPDATE_FUNCTION_NAME)
-            var['Environment']['Variables']['DUMMY'] = random_str(20)
-            client.update_function_configuration(FunctionName=UPDATE_FUNCTION_NAME,
-                                                 Environment=var['Environment'])
-            time.sleep(10)
+            if not is_time_between( TIME_SLEEP_FROM , TIME_TILL_ASLEEP ):
+                response = client.invoke(FunctionName=TRIGGER_FUNCTION_NAME,
+                            InvocationType='Event',Payload='')
+                var = client.get_function_configuration(FunctionName=UPDATE_FUNCTION_NAME)
+                var['Environment']['Variables']['DUMMY'] = random_str(20)
+                client.update_function_configuration(FunctionName=UPDATE_FUNCTION_NAME,
+                                                    Environment=var['Environment'])
+                time.sleep(10)
```

Comment:
Would it be so bad if we got rid of the `else` block entirely, added a `time.sleep(10)` after the if, and let the cron job keep running every 10s? There's not much processing that goes on during each check, and the lesser code there is, the lesser the changes of encountering bugs :v:
@abizerlokhandwala

Reply:
I agree with @Saunved
Waking up every 10 seconds has no cons in terms of processing metrics, and it would also keep our code neat and simple to understand.
@ashawe let's follow this, thanks!

Reply:
I've removed the else block